From 27f3bd2a07a422ec5122cbb3eb0b8bcb7b3a03ab Mon Sep 17 00:00:00 2001
From: Howard Tang <howardt@nvidia.com>
Date: Fri, 18 Jul 2025 13:50:45 -0700
Subject: [PATCH] iio:adc:ti-ads7142:Fixed device id in probe function

Fixed incorrectly using dev to obtain driver_data by properly obtaining
the device id from client. This was previously passed in as an argument
but have been removed in recent update.

Fixes NVBug https://nvbugspro.nvidia.com/bug/5405343

Signed-off-by: Howard Tang <howardt@nvidia.com>
Upstream-Status: Pending [Not submitted to upstream yet]
---
 drivers/iio/adc/ti-ads7142.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/iio/adc/ti-ads7142.c b/drivers/iio/adc/ti-ads7142.c
index 9ab8992e80b7..472c83eb9e5c 100644
--- a/drivers/iio/adc/ti-ads7142.c
+++ b/drivers/iio/adc/ti-ads7142.c
@@ -227,9 +227,9 @@ static int ads7142_read_raw(struct iio_dev *indio_dev,
 
 static int ads7142_probe(struct i2c_client *client)
 {
+	const struct i2c_device_id *id = i2c_client_get_device_id(client);
 	struct iio_dev *indio_dev;
 	struct ads7142_state *state;
-	struct device *dev = &client->dev;
 	int ret;
 
 	indio_dev = devm_iio_device_alloc(&client->dev, sizeof(*state));
@@ -243,11 +243,11 @@ static int ads7142_probe(struct i2c_client *client)
 
 	state->chip = device_get_match_data(&client->dev);
 	if (!state->chip)
-		state->chip = (const struct ads7142_chip_data *)dev->driver_data;
+		state->chip = (const struct ads7142_chip_data *)id->driver_data;
 	if (!state->chip)
 		return dev_err_probe(&client->dev, -EINVAL, "Unknown chip.\n");
 
-	indio_dev->name = "ads7142";
+	indio_dev->name = id->name;
 	indio_dev->channels = state->chip->channels;
 	indio_dev->num_channels = state->chip->num_channels;
 	indio_dev->info = state->chip->info;
-- 
2.43.0

